kong/db/declarative/init.lua

```diff
@@ -398,6 +398,22 @@ function declarative.load_into_db(entities, meta)
 end
 
 
+local function begin_transaction(db)
+  if db.strategy == "postgres" then
+    db.connector:connect("read")
+    db.connector:query("BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ READ ONLY;", "read")
+  end
+end
+
+
+local function commit_transaction(db)
+  if db.strategy == "postgres" then
+    db.connector:query("COMMIT;", "read")
```

Comment:
```suggestion
    db.connector:query("ABORT;", "read")
```

Since we will never make any changes to the data, `ABORT` seems to be better here.

Reply:
@dndx, I googled this a bit before I decided to go with `COMMIT`. Some discussions here:
https://stackoverflow.com/questions/309834/should-i-commit-or-rollback-a-read-transaction

I am fine with both.

Reply:
I changed it to `ROLLBACK` and added a comment about it.